application/apps/indexer/parsers/src/someip.rs

```diff
@@ -0,0 +1,330 @@
+        entry.instance_id,
+        entry.eventgroup_id,
+        entry.major_version,
+        match entry.has_ttl() {
+            true => format!(", Ttl {}", entry.ttl),
+            false => String::from(""),
+        }
+    )
+}
+
+fn sd_option_string(option: &SdEndpointOption) -> String {
+    format!(
+        "{}:{} ({})",
+        option.ip,
+        option.port,
+        match option.proto {
+            IpProto::UDP => String::from("UDP"),
+            IpProto::TCP => String::from("TCP"),
+        },
+    )
+}
+
+type RpcPayloadDetails = (Option<String>, Option<String>, Option<Box<dyn SOMType>>);
+
+fn rpc_payload_details(model: &FibexModel, header: &Header) -> RpcPayloadDetails {
+    let service_id = header.message_id.service_id as usize;
+    let service_version = header.interface_version as usize;
+    let method_id = header.message_id.method_id as usize;
+    let message_type = header.message_type;
+
+    let mut service_name = None;
+    let mut method_name = None;
+
+    let fibex_type = match model.get_service(service_id, service_version) {
+        Some(service) => {
+            service_name = Some(service.name.clone());
+            match service.get_method(method_id) {
+                Some(method) => {
+                    method_name = Some(method.name.clone());
+                    match message_type {
+                        MessageType::Request
+                        | MessageType::RequestNoReturn
+                        | MessageType::Notification => method.get_request(),
+                        MessageType::Response => method.get_response(),
+                        _ => None,
+                    }
+                }
+                _ => None,
+            }
+        }
+        _ => None,
+    };
+
+    let som_type = match fibex_type {
+        Some(value) => match FibexTypes::build(value) {
+            Ok(result) => Some(result),
+            _ => None,
+        },
+        _ => None,
+    };
+
+    (service_name, method_name, som_type)
+}
+
+fn rpc_payload_string(details: RpcPayloadDetails, payload: &RpcPayload) -> String {
+    let (service_name, method_name, som_type) = details;
+
+    let signature = format!(
+        "{}::{}",
+        match service_name {
+            Some(value) => value,
+            _ => String::from("'Unknown Service'"),
+        },
+        match method_name {
+            Some(value) => value,
+            _ => String::from("'Unknown Method'"),
+        }
+    );
+
+    if payload.is_empty() {
+        return format!("- {}", signature);
+    }
+
+    let content = match som_type {
+        Some(mut value) => {
+            let mut parser = SOMParser::new(payload);
+            match value.parse(&mut parser) {
+                Ok(_) => {
+                    format!("{}", value)
+                }
+                Err(error) => format!("{}", error),
+            }
+        }
+        _ => String::from("'Unknown Type'"),
+    };
+
+    format!("- {} {}", signature, content)
```

Comment:
best to write the whole expression in 1 `format!` expression so that we do not create unnecessary copies

Reply:
done.